salt/grains/metadata.py

```diff
@@ -57,6 +66,9 @@ def _search(prefix="latest/"):
     for line in body.split("\n"):
         if line.endswith("/"):
             ret[line[:-1]] = _search(prefix=os.path.join(prefix, line))
+        elif line.endswith(("user-data")):
+            retdata = http.query(os.path.join(HOST, prefix, line)).get("body", None)
+            ret[line] = retdata
```

Comment:
Clarification on what this is for?

Reply:
this is to make sure user-data will be returned correctly, since some user-data content will be cut by this condition:
        elif "=" in line:
            key, value = line.split("=")
            ret[value] = _search(prefix=os.path.join(prefix, key))

Reply:
will remove this fix from commit and re-submit it with a dedicated request, thanks.